### PR TITLE
#20638 fix on update sites

### LINKED ISSUE #20638: Update Site do not update  aliases
Update Site does not update  aliases

STR:
1- Create a new Site without aliases.
2- Run the following REST API call to update the site and add some aliases.
```
PUT {{serverURL}}/api/v1/site?id={{siteId}}

{
    "siteName": "{siteName}",
    "aliases":"postman.hostalias1.com postman.hostalias2.com"
}
```
3- Check that the aliases are not null.

### PR DESCRIPTION
**Pull request by jdotcms**
Fix on update sites

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java [hunk 1]
```diff
@@ -897,6 +899,9 @@ public Response updateSite(@Context final HttpServletRequest httpServletRequest,
             throw new NotFoundException("Site: " + siteIdentifier + " does not exists");
         }
 
+        if (null != site.getMap().get(Contentlet.NULL_PROPERTIES)) {
+            Set.class.cast(site.getMap().get(Contentlet.NULL_PROPERTIES)).clear();
+        }
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java:904]: There's a Contentlet.cleanUp public Method that accomplishes what you need.

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java:904]: Maybe that clean-up logic should take place within the helper .. don't you think?

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java [hunk 2]
```diff
@@ -431,6 +433,18 @@ public Host save(Host host, User user, boolean respectFrontendRoles) throws DotS
         contentletHost.setBoolProperty(Contentlet.DISABLE_WORKFLOW, true);
         contentletHost = APILocator.getContentletAPI().checkin(contentletHost, user, respectFrontendRoles);
 
+        if (!contentletHost.isNew() &&
+                !contentletHost.getTitle().equals(host.get(Host.HOST_NAME_KEY))) {
+            UpdateContainersPathsJob.triggerUpdateContainersPathsJob(
+                    host.get(Host.HOST_NAME_KEY).toString(),
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:439]: is it safe to do host.get(Host.HOST_NAME_KEY).toString() 
NPE???

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java:904]: thanks change done

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**